### ISSUE TITLE
Jump to tabs

### ISSUE THREAD

**Issue by N3C2L** [madprops/grasshopper#113]
Hi,
I hope this is a small feature request but I would be greatly appreciated if you could look into this idea some day.
I am thinking something for an option like "tab jumps with color assignment".
For example: right click a tab and assign a tab jumper number and a specific color. 
Idea is, that the tab list wont be filtered for this tab jump, only the tab with the tab jump number will be focused in current (all) view.
Advantage would be here, that we could quickly jump to specific tabs, for example when these jump tabs will be clicked from the favorite bar.
I hope you can implement this and wish you a great day.

**Comment by madprops** [comment]
Hi,

Would creating a Jump command for each color help?

For instance we can have `Jump Red`, `Jump Green`, etc

Which you can place in the Favorites bar.

**Comment by madprops** [comment]
I just implemented jumps for those 3 special tags and for each color. But I'll wait a bit for further feedback before uploading.

**Comment by madprops** [comment]
> Can we create new colors, assign this color to a tab and then jump to it afterwards?

Yes! And the command icon for jump is the color itself.

Just uploaded `v3000` which includes the jump features. 

Also added in `More` a setting called `Jump Unloaded` (default true) to determine if unloaded tabs should be ignored when jumping or not, by default they are considered.

**Comment by N3C2L** [comment]
Amazing !It's really fun to jump through the tabs like a grasshopper😁 
Thanks also for the setting to jump to unloaded tabs! 
I've thought of one last important point about this new jump feature: (I promise it's the last one for this feature request 😁 ) 
It would be really nice if we could also insert these new jump to color/tag commands here in "Settings > Filter > Favorite Filters". Because then we could quickly jump to the colored tabs with the mouse wheel. This would be fantastic.

**Comment by madprops** [comment]
> It would be really nice if we could also insert these new jump to color/tag commands here in "Settings > Filter > Favorite Filters". Because then we could quickly jump to the colored tabs with the mouse wheel. This would be fantastic. (The "Favorites Filters" setting could then maybe renamed to "Favorite Filters & Jumps")

Right now those commands are limited to filter commands, because the wheel is used to cycle through those filter modes. I don't see how a command like jump would make sense here, or any other command that is not a filter. Because one mousewheel-down you are in jump red, another mousewheel-down and you are in jump-green, doesn't make much sense.

What I can do instead is add 4 new commands to `Favorites`:

`Mousewheel Up Top`,
`Mousewheel Down Top`,

`Mousewheel Up Bottom`,
`Mousewheel Down, Bottom`

Right now those simply scroll the container, but if they're mappable you can then change the action to whatever.

**Comment by madprops** [comment]
It's going to break your favorite commands now since I changed the command names btw. So you'll have to re-add those sorry. Uploaded v3002 just now.

**Comment by N3C2L** [comment]
Thank you. It's really cool that we can now jump each color up or down. 
Currently, for me, each header tab is assigned a different color. This way, I can quickly switch between topic areas when I have these colors as jump colors in the favorites bar. That's why it would be really great if we could also quickly jump between different colored tabs/header tabs. I had thought of two possible options:

Jump color down/up (any colorized tab)
Jump header down/up (any tab that is a header)

I would then put these into your cool new favorites bar mousewheel command list.
It would be really great if this were possible.

**Comment by madprops** [comment]
Added several new jump commands to jump to headers, subheaders, splits, or zones in general.

Also added other commands including holding 'alt' when using up/down arrow to select skipping unloaded.

Plus other fixes. 

I'm gonna close this now but feel free to comment if you find bugs.

**Comment by N3C2L** [comment]
Thanks for these great commands! 👍 ❤️ 
 I noticed a small bug: When I have activated "First" in the Zones settings > Header Action, the "jump header up" command doesn't always work. I have to press "jump header down" several times beforehand, but after that, "jump header up" only goes to the one header where I was last, but not to another header tab above this header tab.

**Comment by madprops** [comment]
> I noticed a small bug

Indeed. This should be fixed on all header actions now.

Also added a `Jump Headers` (with an s)  , which counts headers and subheaders (not splits)

Plus others mods and fixes on `v3011`

**Comment by N3C2L** [comment]
Thanks!
But I think the new "jump headers up" cmd now has the same bug as I described above 😁

**Comment by madprops** [comment]
Yes, thank you for catching that. Also affected `zone` I think. Both fixed on `v3014`

🦗 🦗 🦗 🦗 🦗

**Comment by N3C2L** [comment]
Thanks!

**Comment by N3C2L** [comment]
I think there is another bug in headers up/down commands:
Steps to repro:
Settings > Zones > Header Action > Activate First
Settings > More > Jump unloaded: uncheck
Create some headers and then some unloaded items below these headers
Try to use the headers up/down commands and in my case it's not switching to it.

**Comment by madprops** [comment]
Hmm. I created some headers. Put action to `First`. And added Headers Down/Up to favorites bar.

It seems to work, if an unloaded tab is the first one it activates it.

Maybe I'm missing something.

![4j4j](https://github.com/user-attachments/assets/80c4bd21-4325-4a6c-8d54-f8ed1163b8a2)

**Comment by N3C2L** [comment]
Thanks for checking and I know what was my problem: I had to recreate my old header tabs (the header tabs that I created months ago). But I just duplicated them. And now it's working sometimes. But sometimes when I unload a tab below a header, the tab get's reactivated when jumping with headers/up down.

**Comment by madprops** [comment]
I guess in this case the first tab should be skipped if Jump Unloaded is unchecked, and instead pick the next available tab.

**Comment by N3C2L** [comment]
Yes. Is this working on your side normally?

**Comment by madprops** [comment]
I'm implementing this right now. Though I don't know why you had to remake the headers or what that fixed

**Comment by N3C2L** [comment]
Yeah, don't know it either, sorry. But you can reproduce it by restarting the browser. You have to recreate the header tabs to make it work.

**Comment by madprops** [comment]
Oh so it was ignoring the headers because they were unloaded it seems

**Comment by N3C2L** [comment]
Yes, do you think it's possible to load the header when switching headers and switch afterwards to the tab below?

**Comment by madprops** [comment]
Yeah I'll look into it. Ideally they should be loaded always

**Comment by madprops** [comment]
Alright lots of changes to the jump function, header jumping should be better

**Comment by N3C2L** [comment]
Great! Can confirm it's working, even when browser is restarted! Thanks for this!

**Comment by madprops** [comment]
Sure just let me know when it breaks haha

**Comment by N3C2L** [comment]
Will do 😁
